src/components/StatesButton/StatesButton.tsx

```diff
@@ -3,39 +3,57 @@ import * as classNames from 'classnames';
 import style from './StatesButton.st.css';
 import { Button, ButtonProps } from '../Button';
 import { TPAComponentProps } from '../../types';
-
-const delay = time => new Promise(resolve => setTimeout(resolve, time));
+import { BUTTON_STATES } from './constants';
+import Timeout = NodeJS.Timeout;
 
 export interface StatesButtonProps extends ButtonProps, TPAComponentProps {
+  state: BUTTON_STATES;
   disabled: boolean;
-  text: string;
+  idleContent: string | React.ReactElement;
+  failureContent?: string | React.ReactElement;
+  inProgressContent?: string | React.ReactElement;
+  successContent?: string | React.ReactElement;
+  onSuccessEnd?: Function;
   dataHook?: string;
```

Comment:
While we're at it, you can remove the `dataHook` prop.
The `TPAComponentProps` type already introduces a `data-hook` prop (which is the preferred syntax).